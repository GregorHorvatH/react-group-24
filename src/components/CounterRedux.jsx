import { connect } from 'react-redux';
import {
  incrementAction,
  decrementAction,
  setStepAction,
} from '../reducer/counter/actions';

const CounterRedux = ({ value, step, increment, decrement, setStep }) => {
  return (
    <div className="counter-redux">
      <h4>Counter Redux</h4>
      <select value={step} onChange={(e) => setStep(Number(e.target.value))}>
        <option value="1">1</option>
        <option value="5">5</option>
        <option value="10">10</option>
        <option value="50">50</option>
      </select>

      <button onClick={() => decrement(step)}>-</button>
      <span>{value}</span>
      <button onClick={() => increment(step)}>+</button>
    </div>
  );
};

const mapStateToProps = (state) => ({
  value: state.counter.value,
  step: state.counter.step,
});

const mapDispatchToProps = {
  increment: incrementAction,
  decrement: decrementAction,
  setStep: setStepAction,
};

export default connect(mapStateToProps, mapDispatchToProps)(CounterRedux);
