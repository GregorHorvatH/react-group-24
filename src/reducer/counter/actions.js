// import types from './types';

// export const incrementAction = (payload) => ({
//   type: types.COUNTER_INCREMENT,
//   payload,
// });

// export const decrementAction = (payload) => ({
//   type: types.COUNTER_DECREMENT,
//   payload,
// });

// export const setStepAction = (payload) => ({
//   type: types.COUNTER_SET_STEP,
//   payload,
// });

import { createAction } from '@reduxjs/toolkit';

export const incrementAction = createAction('counter/increment');
export const decrementAction = createAction('counter/decrement');
export const setStepAction = createAction('counter/set_step');

// export const someAction = createAction('some/action', (name, age) => ({
//   payload: {
//     id: Date.now(),
//     name,
//     age,
//   },
// }));
