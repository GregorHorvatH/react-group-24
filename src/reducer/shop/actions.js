// ===== before =====
// import types from './types';

// export const shopFilterSet = (payload) => ({
//   type: types.SHOP_FILTER_SET,
//   payload,
// });

// export const shopCartAdd = (payload) => ({
//   type: types.SHOP_CART_ADD,
//   payload,
// });

// export const shopCartDelete = (payload) => ({
//   type: types.SHOP_CART_DELETE,
//   payload,
// });

// export const cartIncrement = (payload) => ({
//   type: types.CART_INCREMENT,
//   payload,
// });

// export const cartDecrement = (payload) => ({
//   type: types.CART_DECREMENT,
//   payload,
// });

// ===== after =====
import { createAction } from '@reduxjs/toolkit';

export const shopFilterSet = createAction('shop/filter/set');
export const shopCartAdd = createAction('shop/cart/add');
export const shopCartDelete = createAction('shop/cart/delete');
export const cartIncrement = createAction('cart/increment');
export const cartDecrement = createAction('cart/decrement');

export const shopItemsRequest = createAction('shop/items/request');
export const shopItemsSuccess = createAction('shop/items/success');
export const shopItemsFailure = createAction('shop/items/failure');

export const cartItemsRequest = createAction('shop/cart/request');
export const cartItemsSuccess = createAction('shop/cart/success');
export const cartItemsFailure = createAction('shop/cart/failure');
