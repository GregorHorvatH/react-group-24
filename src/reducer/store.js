import { configureStore } from '@reduxjs/toolkit';
import { persistStore } from 'redux-persist';
// import storage from 'redux-persist/lib/storage'; // defaults to localStorage for web
import rootReducer from './reducer';

// const persistConfig = {
//   key: 'user',
//   storage,
//   whitelist: ['user'],
// };

export const store = configureStore({
  // reducer: persistReducer(persistConfig, rootReducer),
  reducer: rootReducer,
});

export const persistor = persistStore(store);

// export default store;
