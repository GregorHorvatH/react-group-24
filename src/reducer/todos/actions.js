import types from './types';

export const addTodo = (todo) => ({
  type: types.TODO_ADD,
  payload: todo,
});

export const deleteTodo = (id) => ({
  type: types.TODO_DELETE,
  payload: id,
});

export const toggleTodo = (id) => ({
  type: types.TODO_TOGGLE,
  payload: id,
});

export const selectTodo = (id) => ({
  type: types.TODO_SELECT,
  payload: id,
});

export const editTodo = (payload) => ({
  type: types.TODO_EDIT,
  payload,
});

export const toggleModal = () => ({
  type: types.MODAL_TOGGLE,
});
