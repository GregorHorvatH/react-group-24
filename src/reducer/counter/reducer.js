// import { combineReducers } from 'redux';
// import types from './types';
import { createReducer, combineReducers } from '@reduxjs/toolkit';
import * as actions from './actions';

const initialState = {
  value: 1,
  step: 1,
};

const value = createReducer(initialState.value, {
  [actions.incrementAction.type]: (state, action) =>
    state + (action.payload || 1),
  [actions.decrementAction.type]: (state, action) =>
    state - (action.payload || 1),
});

const step = createReducer(initialState.step, {
  [actions.setStepAction.type]: (_, action) => action.payload,
});

// const counterReducer = (state = initialState, action) => {
//   switch (action.type) {
//     case types.COUNTER_INCREMENT:
//       return {
//         ...state,
//         value: state.value + action.payload,
//       };

//     case types.COUNTER_DECREMENT:
//       return {
//         ...state,
//         value: state.value - action.payload,
//       };

//     case types.COUNTER_SET_STEP:
//       return {
//         ...state,
//         step: action.payload,
//       };

//     default:
//       return state;
//   }
// };

// export default counterReducer;

// const value = (state = initialState.value, action) => {
//   switch (action.type) {
//     case types.COUNTER_INCREMENT:
//       return state + action.payload;

//     case types.COUNTER_DECREMENT:
//       return state - action.payload;

//     default:
//       return state;
//   }
// };

// const step = (state = initialState.step, action) => {
//   switch (action.type) {
//     case types.COUNTER_SET_STEP:
//       return action.payload;

//     default:
//       return state;
//   }
// };

export default combineReducers({
  value,
  step,
});
