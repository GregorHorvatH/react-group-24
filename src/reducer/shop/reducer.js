// ===== after =====
import { createReducer, combineReducers } from '@reduxjs/toolkit';
import * as actions from './actions';

const initialState = {
  items: [
    // {
    //   id: 1,
    //   name: 'phone 1',
    //   price: 5000,
    //   description:
    //     'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Pariatur ab optio explicabo cumque ut! Provident quasi, explicabo perferendis nobis vero pariatur architecto culpa laudantium praesentium perspiciatis tenetur commodi ex similique.',
    //   count: 5,
    // },
    // {
    //   id: 2,
    //   name: 'phone 2',
    //   price: 7000,
    //   description:
    //     'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Pariatur ab optio explicabo cumque ut! Provident quasi, explicabo perferendis nobis vero pariatur architecto culpa laudantium praesentium perspiciatis tenetur commodi ex similique.',
    //   count: 3,
    // },
    // {
    //   id: 3,
    //   name: 'phone 3',
    //   price: 8000,
    //   description:
    //     'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Pariatur ab optio explicabo cumque ut! Provident quasi, explicabo perferendis nobis vero pariatur architecto culpa laudantium praesentium perspiciatis tenetur commodi ex similique.',
    //   count: 1,
    // },
    // {
    //   id: 4,
    //   name: 'notebook 1',
    //   price: 11000,
    //   description:
    //     'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Pariatur ab optio explicabo cumque ut! Provident quasi, explicabo perferendis nobis vero pariatur architecto culpa laudantium praesentium perspiciatis tenetur commodi ex similique.',
    //   count: 2,
    // },
  ],
  cart: [
    // {
    //   id: 1,
    //   productId: 2,
    //   count: 1,
    // },
    // {
    //   id: 2,
    //   productId: 4,
    //   count: 1,
    // },
  ],
  filter: {},
  isLoading: false,
  cartIsLoading: false,
};

// const items = (state = initialState.items, action) => state;

// ===== before =====
// import { combineReducers } from 'redux';
// import types from './types';

// const cart = (state = initialState.cart, action) => {
//   switch (action.type) {
//     case types.SHOP_CART_ADD:
//       return [...state, action.payload];

//     case types.SHOP_CART_DELETE:
//       return state.filter(({ productId }) => productId !== action.payload);

//     case types.CART_INCREMENT:
//       return state.map((item) =>
//         item.id === action.payload
//           ? {
//               ...item,
//               count: item.count + 1,
//             }
//           : item,
//       );

//     case types.CART_DECREMENT:
//       return state.map((item) =>
//         item.id === action.payload
//           ? {
//               ...item,
//               count: item.count - 1 < 0 ? 0 : item.count - 1,
//             }
//           : item,
//       );

//     default:
//       return state;
//   }
// };

// const filter = (state = initialState.filter, action) => {
//   switch (action.type) {
//     case types.SHOP_FILTER_SET:
//       return action.payload;

//     default:
//       return state;
//   }
// };

const shopCartAdd = (state, action) => [...state, action.payload];

const shopCartDelete = (state, action) =>
  state.filter(({ productId }) => productId !== action.payload);

const cartIncrement = (state, action) =>
  state.map((item) =>
    item.id === action.payload
      ? {
          ...item,
          count: item.count + 1,
        }
      : item,
  );

const cartDecrement = (state, action) =>
  state.map((item) =>
    item.id === action.payload
      ? {
          ...item,
          count: item.count - 1 < 0 ? 0 : item.count - 1,
        }
      : item,
  );

const items = createReducer(initialState.items, {
  [actions.shopItemsSuccess.type]: (_, action) => action.payload,
});

const isLoading = createReducer(initialState.isLoading, {
  [actions.shopItemsRequest.type]: () => true,
  [actions.shopItemsSuccess.type]: () => false,
  [actions.shopItemsFailure.type]: () => false,
});

const cartIsLoading = createReducer(initialState.cartIsLoading, {
  [actions.cartItemsRequest.type]: () => true,
  [actions.cartItemsSuccess.type]: () => false,
  [actions.cartItemsFailure.type]: () => false,
});

const cart = createReducer(initialState.cart, {
  [actions.cartItemsSuccess.type]: (_, action) => action.payload,
  [actions.shopCartAdd.type]: shopCartAdd,
  [actions.shopCartDelete.type]: shopCartDelete,
  [actions.cartIncrement.type]: cartIncrement,
  [actions.cartDecrement.type]: cartDecrement,
});

const filter = createReducer(initialState.filter, {
  [actions.shopFilterSet.type]: (_, action) => action.payload,
});

export default combineReducers({
  items,
  cart,
  filter,
  isLoading,
  cartIsLoading,
});
