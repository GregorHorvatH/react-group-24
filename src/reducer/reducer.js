import { combineReducers } from 'redux';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage'; // defaults to localStorage for web

import counter from './counter/reducer';
import todos from './todos/reducer';
import shop from './shop/reducer';
import user from './user/reducer';

const persistConfig = {
  key: 'token',
  storage,
  whitelist: ['token'],
};

export default combineReducers({
  counter,
  todos,
  shop,
  user: persistReducer(persistConfig, user),
});
