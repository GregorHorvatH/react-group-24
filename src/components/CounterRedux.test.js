import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../reducer/store';
import { incrementAction } from '../reducer/counter/actions';
import reducer from '../reducer/counter/reducer';

import CounterRedux from './CounterRedux';

describe('CounterRedux', () => {
  it('should render', () => {
    const { container } = render(
      <Provider store={store}>
        <CounterRedux />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('incrementAction should return a new object', () => {
    expect(incrementAction(123)).toMatchObject({
      type: 'counter/increment',
      payload: 123,
    });
  });

  it('reducer', () => {
    const initialState = {
      value: 1,
      step: 1,
    };

    const action = { type: 'counter/increment' };
    const newState = reducer(initialState, action);

    expect(newState).toMatchObject({
      value: 2,
      step: 1,
    });
  });
});
