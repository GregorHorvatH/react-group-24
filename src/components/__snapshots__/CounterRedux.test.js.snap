// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterRedux should render 1`] = `
<div>
  <div
    class="counter-redux"
  >
    <h4>
      Counter Redux
    </h4>
    <select>
      <option
        value="1"
      >
        1
      </option>
      <option
        value="5"
      >
        5
      </option>
      <option
        value="10"
      >
        10
      </option>
      <option
        value="50"
      >
        50
      </option>
    </select>
    <button>
      -
    </button>
    <span>
      1
    </span>
    <button>
      +
    </button>
  </div>
</div>
`;
